src/main/java/org/opensearch/neuralsearch/processor/rerank/context/QueryContextSourceFetcher.java

```diff
@@ -76,9 +78,10 @@ public void fetchContext(
                     throw new IllegalArgumentException(
                         String.format(
                             Locale.ROOT,
-                            "%s exceeded the maximum path length of %d",
+                            "%s exceeded the maximum path length of %d nested fields or %d characters",
```

Comment:
I think we need to keep one exception per one failed validation. Can we make throwing exception a part of `validatePath` method, and throw exception with specific reason for each of two cases? 
One more ask - please add log at `error` level with exact value entered by user.